js/dydx.js

```diff
@@ -0,0 +1,973 @@
+        };
+        let query = undefined;
+        if (method === 'GET') {
+            query = this.urlencode (params);
+            url = url + '?' + query;
+        } else {
+            body = this.json (params);
+        }
+        const timestamp = this.milliseconds ();
+        if (api === 'private') {
+            if (method === 'GET') {
+                payload = query;
+            } else {
+                payload = body;
+            }
+            if (path === 'onboarding') {
+                if (method === 'POST') {
+                    // onboarding endpoint: POST /v3/onboarding
+                    headers['DYDX-ETHEREUM-ADDRESS'] = this.ethereumAddress;
+                    payload['action'] = 'DYDX-ONBOARDING';
+                    payload['onlySignOn'] = 'https://trade.dydx.exchange';
+                    const signature = this.hmac (this.encode (payload), this.encode (this.secret));
+                    headers['DYDX-SIGNATURE'] = signature; // EIP-712-compliant Ethereum signature
+                }
+            } else if (method === 'DELETE') {
+                if (path === 'api-keys') {
+                    // Ethereum Key Private Endpoints: POST, DELETE /v3/api-keys
+                    headers['DYDX-TIMESTAMP'] = timestamp;
+                    headers['DYDX-ETHEREUM-ADDRESS'] = this.ethereumAddress;
+                    payload['method'] = 'GET|POST';
+                    payload['requestPath'] = '/v3/api-keys';
+                    payload['body'] = ''; // empty for GET and DELETE
+                    payload['timestamp'] = timestamp;
+                    const signature = this.hmac (this.encode (payload), this.encode (this.secret));
+                    headers['DYDX-SIGNATURE'] = signature; // EIP-712-compliant Ethereum signature
+                }
+            } else {
+                // All other API Key Private Endpoints
+                headers['DYDX-TIMESTAMP'] = timestamp;
+                headers['DYDX-ETHEREUM-ADDRESS'] = this.ethereumAddress;
+                headers['DYDX-PASSPHRASE'] = this.passPhrase;
+                const signature = this.hmac (this.encode (payload), this.encode (this.secret));
+                headers['DYDX-SIGNATURE'] = signature; // SHA-256 HMAC produced as described below, and encoded as a Base64 string
+            }
+        }
+        return { 'url': url, 'method': method, 'body': payload, 'headers': headers };
+    }
+
+    handleErrors (code, reason, url, method, headers, body, response, requestHeaders, requestBody) {
+        if (response === undefined) {
+            return; // fallback to default error handler
+        }
+        const errorCode = this.safeString (response, 'code');
+        const message = this.safeString (response, 'error', '');
+        this.throwExactlyMatchedException (this.exceptions['exact'], message, this.id + ' ' + message);
+        this.throwExactlyMatchedException (this.exceptions['exact'], errorCode, this.id + ' ' + message);
+    }
```

Comment:
Add an exception for `this.throwBroadlyMatchedException`

Reply:
![image](https://user-images.githubusercontent.com/1892654/122624560-00671600-d0a1-11eb-9cc2-a537e5678ddf.png)
@mrbaguvix hers' a screenshot of my debug console.

Reply:
oh, and here's the text if that's easier to read:

```
$ node run-tests --js dydx
Testing { exchanges: ["dydx"], symbol: "all", keys: { '--js': true, '--php': false, '--python': false, '--python-async': false, '--php-async': false }, maxConcurrency: 5 } (run-tests.js:255)
[100%] Testing dydx [ExchangeNotAvailable] [ExchangeNotAvailable] (testExchange @ run-tests.js:172)

WARN dydx (JavaScript): (explain @ run-tests.js:191)

	
	TESTING { exchange: "dydx", symbol: "all" }  (<anonymous> @ test.js:30)
	
	17 symbols [] (loadExchange @ test.js:183)
	SYMBOL: 1INCH-USD (testExchange @ test.js:270)
	fetching currencies not supported (exports [as fetchCurrencies] @ test.fetchCurrencies.js:37)
	1INCH-USD fetchTicker () not supported (exports [as fetchTicker] @ test.fetchTicker.js:53)
	fetching all tickers at once not supported (exports [as fetchTickers] @ test.fetchTickers.js:51)
	[ExchangeNotAvailable] dydx Not Found (tryAllProxies @ test.js:377)
	17 symbols [] (loadExchange @ test.js:183)
	SYMBOL: 1INCH-USD (testExchange @ test.js:270)
	fetching currencies not supported (exports [as fetchCurrencies] @ test.fetchCurrencies.js:37)
	1INCH-USD fetchTicker () not supported (exports [as fetchTicker] @ test.fetchTicker.js:53)
	fetching all tickers at once not supported (exports [as fetchTickers] @ test.fetchTickers.js:51)
	[ExchangeNotAvailable] dydx GET https://cors-anywhere.herokuapp.com/https://api.stage.dydx.exchange/v3/candles?market=1INCH-USD&resolution=1h&fromISO=1624024270823&limit=10 403 Forbidden See /corsdemo for more info (tryAllProxies @ test.js:377) (explain @ run-tests.js:193)
	
WARN ["dydx"] (run-tests.js:272)
All done, 1 warnings (run-tests.js:276)
```

Reply:
Thanks for pointing me in the right direction. I've now fixed is